app/api/controllers/RequestHandler.scala

```diff
@@ -109,19 +109,28 @@ object RequestHandler {
           message = s"[${ctx.endpointLogContext.controllerName}][${ctx.endpointLogContext.endpointName}] " +
             s"with correlationId : ${ctx.correlationId}")
 
-        val result =
-          for {
-            parsedRequest   <- EitherT.fromEither[Future](parser.parseRequest(rawData))
-            serviceResponse <- EitherT(service(parsedRequest))
-          } yield doWithContext(ctx.withCorrelationId(serviceResponse.correlationId)) { implicit ctx: RequestContext =>
-            handleSuccess(rawData, parsedRequest, serviceResponse)
-          }
-
-        result.leftMap { errorWrapper =>
-          doWithContext(ctx.withCorrelationId(errorWrapper.correlationId)) { implicit ctx: RequestContext =>
+        val maybeGovTestScenario = ctx.hc.otherHeaders.find(header => header._1 == "Gov-Test-Scenario").map(headers => headers._2).getOrElse("")
+        if (maybeGovTestScenario == "REQUEST_CANNOT_BE_FULFILLED") {
```

Comment:
Does this check the environment somewhere?

Reply:
(should only accept the REQUEST_CANNOT_BE_FULFILLED if it's a non-e2e env, basically either local or ET)

Reply:
In fact it would be better as a config setting, so the feature can be enabled/disabled specifically in each env.